yb-voyager/cmd/assessMigrationCommand.go

```diff
@@ -924,21 +928,82 @@ func addAssessmentIssuesForRedundantIndexes() error {
-			return fmt.Errorf("error scanning rows for redundant indexes: %w", err)
+			return nil, fmt.Errorf("error scanning rows for redundant indexes: %w", err)
 		}
 		redundantIndex.DBType = source.DBType
 		redundantIndexesInfo = append(redundantIndexesInfo, redundantIndex)
 	}
-	redundantIssues := parserIssueDetector.GetRedundantIndexIssues(redundantIndexesInfo)
-	for _, redundantIssue := range redundantIssues {
-		convertedAnalyzeIssue := convertIssueInstanceToAnalyzeIssue(redundantIssue, "", false, false)
-		issue := convertAnalyzeSchemaIssueToAssessmentIssue(convertedAnalyzeIssue, redundantIssue.MinimumVersionsFixedIn)
-		assessmentReport.AppendIssues(issue)
+	return redundantIndexesInfo, nil
+}
+
+func fetchColumnStatisticsInfo() ([]utils.ColumnStatistics, error) {
+	query := fmt.Sprintf(`SELECT schema_name, table_name, column_name, null_frequency, effective_n_distinct, max_frequency, max_frequent_val from %s`,
+		migassessment.COLUMN_STATISTICS)
+	rows, err := assessmentDB.Query(query)
+	if err != nil {
+		return nil, fmt.Errorf("error querying-%s on assessmentDB for column statistics: %w", query, err)
+	}
+	defer func() {
+		closeErr := rows.Close()
+		if closeErr != nil {
+			log.Warnf("error closing rows while fetching column statistics %v", err)
+		}
+	}()
+
+	var columnStats []utils.ColumnStatistics
+	for rows.Next() {
+		var stat utils.ColumnStatistics
+		err := rows.Scan(&stat.SchemaName, &stat.TableName, &stat.ColumnName, &stat.NullFrequency, &stat.DistinctValues, &stat.MaxFrequency, &stat.MaxFrequentValue)
+		if err != nil {
+			return nil, fmt.Errorf("error scanning rows for most frequent values indexes: %w", err)
+		}
+		stat.DBType = source.DBType
+		columnStats = append(columnStats, stat)
+	}
+	return columnStats, nil
+}
+
+func fetchAndParseColumnStatisticsForIndexIssues() error {
+	if source.DBType != POSTGRESQL {
+		return nil
+	}
+	var err error
+	columnStats, err := fetchColumnStatisticsInfo()
+	if err != nil {
+		return fmt.Errorf("error fetching column stats from assessement db: %v", err)
+	}
+	parserIssueDetector.PopulateColumnStatisticsMap(columnStats)
+	return nil
+}
+
+func addAssessmentIssuesForRedundantIndex() error {
+	if source.DBType != POSTGRESQL {
+		return nil
+	}
+	redundantIndexesInfo, err := fetchRedundantIndexInfo()
+	if err != nil {
+		log.Errorf("error fetching redundant index information: %v", err)
+	}
+
+	var redundantIssues []queryissue.QueryIssue
+	redundantIssues = append(redundantIssues, queryissue.GetRedundantIndexIssues(redundantIndexesInfo)...)
+	for _, issue := range redundantIssues {
+
+		convertedAnalyzeIssue := convertIssueInstanceToAnalyzeIssue(issue, "", false, false)
+		convertedIssue := convertAnalyzeSchemaIssueToAssessmentIssue(convertedAnalyzeIssue, issue.MinimumVersionsFixedIn)
+		assessmentReport.AppendIssues(convertedIssue)
 	}
 	return nil
 }
 
 func getAssessmentReportContentFromAnalyzeSchema() error {
+
+	var err error
+	err = fetchAndParseColumnStatisticsForIndexIssues()
```

Comment:
let's add a comment here explaining what we're doing. (i.e. reading column stats form assessment db and setting it on the parserIssueDetector, enabling it to detect performance issues.